### ISSUE TITLE
homebrew_cask should accept an appdir parameter

### ISSUE THREAD

**Issue by siancu** [ansible/ansible-modules-extras#167]
##### Issue Type:

Feature Idea.
##### Ansible Version:

ansible --version
ansible 1.8.2
  configured module search path = None
##### Environment:

I'm running Ansible on OSX 10.10.1 and I'm managing the same computer (local connection).
##### Summary:

By default, cask symlinks the apps to ~/Applications instead of /Applications.

The brew cask command accepts a "--appdir" parameter but with homebrew_cask it is impossible to set it. So I would like to request an improvement to the module to add this parameter.

At the moment I've patched locally the source of the module by hardcoding it to /Applications like this:

```
    cmd = [opt
           for opt in (self.brew_path, 'cask', 'install', '--appdir=/Applications', self.current_cask)
           if opt]
```

(lines 398-400 in homebrew_cask.py). It seems to work so far. But this is not the right solution, obviously.
##### Steps To Reproduce:

For example, the homebrew_cask task could look like this:

```
- name: Install Homebrew cask apps
  homebrew_cask: "name=spotify state=present appdir=/Applications"
```

**Comment by elnappo** [comment]
I use the environment variable `HOMEBREW_CASK_OPTS`. Hope this is good enough for you, I'm fine with this ;)

``` yaml
- name: Install Homebrew cask apps
  homebrew_cask: name=spotify state=present
  environment:
    HOMEBREW_CASK_OPTS: --appdir=/Applications
```

**Comment by budda** [comment]
@elnappo many thanks for the pointer about 'environment' - worked flawlessly!

**Comment by robynbergeron** [comment]
[module: packaging/os/homebrew_cask.py]

**Comment by robynbergeron** [comment]
@siancu Thanks for submitting this feature idea -- and thanks to @elnappo and @budda for your comments here as well. Sorry it's taken us a while to triage it. 

@danieljaouen Pinging you to bring this feature request for your attention. Please note that you are under no particular obligation to add this feature.

We encourage all Ansible community members to submit pull requests to satisfy feature requests. Consult the module guidelines (http://docs.ansible.com/ansible/developing_modules.html) for more information.

**Comment by gregdek** [comment]
Re-pinging new maintainer @indrajitr.

**Comment by indrajitr** [comment]
homebrew_cask ≠ homebrew :) I don't use cask too often, but allow me a couple of days -- I'll get back on this.

**Comment by indrajitr** [comment]
Given that there are [whole lot of options](https://github.com/caskroom/homebrew-cask/blob/master/USAGE.md#options) that can be passed via `HOMEBREW_CASK_OPTS`, it makes sense to pass `--appdir` via `HOMEBREW_CASK_OPTS` as well for consistency in treatment (aligning with other options that can be passed).

Let us keep this feature request open for now. At a future point we can have _all_ the cask options available via module arguments (instead of randomly picking one at a time).

**Comment by indrajitr** [comment]
#544 is a reasonable attempt which is quite generic.

**Comment by indrajitr** [comment]
This is addressed more reasonably in #544. This should be marked resolved once #544 is merged.

**Comment by elnappo** [comment]
from homebrew cask README:

> Important: At some point in the future Homebrew-cask will change its behaviour from linking apps to moving them. See issue [#13201](https://github.com/caskroom/homebrew-cask/issues/13201) for details.

from issue [#13201](https://github.com/caskroom/homebrew-cask/issues/13201):

> apps will be moved to /Applications by default.

**Comment by indrajitr** [comment]
Which is why a more generic version is preferred to this one. That would include additional cask options (and even overriding `appdir` to a different location). Note that `appdir` default is changing, but the option would probably be around.

**Comment by elnappo** [comment]
This should be fixed with https://github.com/ansible/ansible-modules-extras/pull/2086

**Comment by robynbergeron** [comment]
@siancu: this should now be resolved in 2.1 -- thanks for your patience, and thanks to those who helped out with making this happen (@enriclluelles, @indrajitr)

Closing this one :)
